nixos/lib/testing/driver.nix

```
@@ -108,6 +108,12 @@ in
       defaultText = literalMD "set by the test framework";
     };
 
+    makeTestDriver = mkOption {
+      description = mdDoc "Function that returns the package containing the test driver that will perform the testing instrumentation";
+      type = types.functionTo types.package;
```

```suggestion
      type = types.functionTo types.package;
      internal = true;
```

This does not seem like a principled interface that we want to support indefinitely.

(otherwise, needs way more docs)

---

Agreed